src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,34 +34,47 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+        
+        String[] addressParts = splitAddressByComma(address);
+        this.block = new Block(addressParts[ADDRESS_BLOCK_INDEX].trim());
+        this.street = new Street(addressParts[ADDRESS_STREET_INDEX].trim());
+        this.unit = new Unit(addressParts[ADDRESS_UNIT_INDEX].trim());
+        this.postalCode = new PostalCode(addressParts[ADDRESS_POSTAL_CODE_INDEX].trim());
     }
 
     /**
      * Returns true if a given string is a valid person email.
      */
     public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
+        return (splitAddressByComma(test).length == 4) 
+                && test.matches(ADDRESS_VALIDATION_REGEX);
```

Comment:
Good job here using regex to check for validity. Maybe consider replace the magic number with a constant.